src/double64.c

```diff
@@ -282,148 +282,69 @@ double64_init	(SF_PRIVATE *psf)
 
 double
 double64_be_read (const unsigned char *cptr)
-{	int		exponent, negative, upper, lower ;
+{
 	double	dvalue ;
+	uint8_t *dbytes = (uint8_t *) &dvalue ;
 
-	negative = (cptr [0] & 0x80) ? 1 : 0 ;
-	exponent = ((cptr [0] & 0x7F) << 4) | ((cptr [1] >> 4) & 0xF) ;
-
-	/* Might not have a 64 bit long, so load the mantissa into a double. */
-	upper = (((cptr [1] & 0xF) << 24) | (cptr [2] << 16) | (cptr [3] << 8) | cptr [4]) ;
-	lower = (cptr [5] << 16) | (cptr [6] << 8) | cptr [7] ;
-
-	if (exponent == 0 && upper == 0 && lower == 0)
-		return 0.0 ;
-
-	dvalue = upper + lower / ((double) 0x1000000) ;
-	dvalue += 0x10000000 ;
-
-	exponent = exponent - 0x3FF ;
-
-	dvalue = dvalue / ((double) 0x10000000) ;
-
-	if (negative)
-		dvalue *= -1 ;
-
-	if (exponent > 0)
-		dvalue *= pow (2.0, exponent) ;
-	else if (exponent < 0)
-		dvalue /= pow (2.0, abs (exponent)) ;
+	dbytes [0] = cptr [7] ;
+	dbytes [1] = cptr [6] ;
+	dbytes [2] = cptr [5] ;
+	dbytes [3] = cptr [4] ;
+	dbytes [4] = cptr [3] ;
+	dbytes [5] = cptr [2] ;
+	dbytes [6] = cptr [1] ;
+	dbytes [7] = cptr [0] ;
```

Comment:
given that you aren't doing endianness targetting here, why not use `memcpy` instead? GCC/Clang optimize `memcpy` away on arguments with compile-time sizes.